From 9209d5121dfb3049cbf280139c4cc40c2038edcc Mon Sep 17 00:00:00 2001
From: David Herrmann <dh.herrmann@gmail.com>
Date: Tue, 17 Sep 2013 17:40:01 +0200
Subject: [PATCH 141/169] logind: fix session_activate(vtnr = 0)

VT numbers start with 1. If a session has vtnr == 0, we must not assume it
is running on a VT.
Note that this could trigger the assert() below as CreateSession() sets
vtnr to 0, not <0.
---
 src/login/logind-session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/login/logind-session.c b/src/login/logind-session.c
index 407429c..ab1c79c 100644
--- a/src/login/logind-session.c
+++ b/src/login/logind-session.c
@@ -363,7 +363,7 @@ int session_activate(Session *s) {
         assert(s);
         assert(s->user);
 
-        if (s->vtnr < 0)
+        if (s->vtnr <= 0)
                 return -ENOTSUP;
 
         if (!s->seat)
-- 
1.8.4

